FiveCalls/FiveCalls/ContactLog.swift

```diff
@@ -93,10 +93,16 @@ struct ContactLogs {
     static private var filePath: URL {
         let pantryDirName = "com.thatthinginswift.pantry"
         let appSupportDir = NSSearchPathForDirectoriesInDomains(.applicationSupportDirectory, .userDomainMask, true).first!
-        let targetPath = URL(fileURLWithPath: appSupportDir).appendingPathComponent(pantryDirName).appendingPathComponent(ContactLogs.persistenceKey, isDirectory: false)
+
+        let targetPath: URL
+        if UserDefaults.standard.bool(forKey: "isBeingTested") {
+            targetPath = FileManager.default.temporaryDirectory.appendingPathComponent(ContactLogs.persistenceKey, isDirectory: false)
+        } else {
+            targetPath = URL(fileURLWithPath: appSupportDir).appendingPathComponent(pantryDirName).appendingPathComponent(ContactLogs.persistenceKey, isDirectory: false)
+        }
```

Comment:
Funny enough I did a screencast on how I tend to do this sort of thing:

<img width="994" alt="image" src="https://user-images.githubusercontent.com/59140/71325640-47c51400-24b5-11ea-8b41-1c78dec8fd51.png">

https://nsscreencast.com/episodes/341-detect-testing-in-app-delegate

Reply:
Much better - less configuration is better.